src/lib/net/classes/remote.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * !TODO: Add server/client differentation, yeah I know this should have made from the very first start, but lol,
+ * that will take a bit of my brain, but i already got something planned out for this.
+ */
+
+import { BaseEvent } from "../types";
+
+export declare enum EAction {
+	COND = "condition",
+	DO = "do",
+}
+
+function isAFunction(func: (() => boolean) | boolean): func is Callback {
+	return typeOf(func) === "function";
+}
+
+export class RemoteClass<T extends unknown[]> implements BaseEvent<T> {
+	protected listeners: Array<[Callback | boolean, EAction]> = new Array();
+
+	public do<X>(func: (...args: [...T]) => X): BaseEvent<[X]> {
+		this.listeners.push([func, EAction.DO]);
+
+		return this as unknown as BaseEvent<[X]>;
+	}
+
+	public condition<X extends () => boolean | boolean>(condition: X): BaseEvent<T> {
```

Comment:
`X extends () => boolean` only, also, make an alias type for this kind of evaluable condition in the conditions library

Reply:
Sounds good!